### ISSUE TITLE
0 and Nan values 

### ISSUE THREAD

**Issue by MinaJf** [deepmedic/deepmedic#213]
Hi all,
First of thank you so much and congrats for your great job.
I have a 3D dataset with very small size of tumor in each image.
I do not have any ROI mask, I only have the images and the ground truths.
I pre-processed each image by using zero-mean of whole image and fed to the model and I got Nan in the probability map classes and 0 for segmentation in the predictions folder.
I noticed that I have to apply the zero-mean only on ROI not whole images. In my case what I can do (I do not have any ROI mask and besides the tumor is too small)?
Thanks in advance.

**Comment by Kamnitsask** [comment]
Hello, thank you for your interest in our work.

The careful preprocessing is necessary for things to work out.

What part of the anatomy is imaged?

E.g. if it is brain, ROI mask can easily be created by a short script that thresholds out the low intensities of the air (after brainmask ideally).

For certain anatomies ROI masks may not be necessary to create (e.g. if it is abdominal/cardiac imaging).

Have you also divided the intensities with standard deviation during intensity-normalization?
If you are not sure you are doing the normalization correctly, the code can do it. For this, in the trainConfig.cfg that you are using set the following near the very end:

```
norm_zscore_prms = {'apply_to_all_channels': True,   (===> instead of False by default)
                                    'apply_per_channel': None,
                                    'cutoff_percents': [5., 95.],
                                    'cutoff_times_std': [3.,3.],
                                    'cutoff_below_mean': False}
```

**Comment by Kamnitsask** [comment]
A different set of questions, just to double-check:

Have you successfully trained the TinyCNN model?
What version of tensorflow are you using?

(I am wondering if certain TF versions are not compatible with the code and can cause your problem, I am not sure. If the TinyCNN model works, then TF version should be fine.)

**Comment by MinaJf** [comment]
Thanks for your response. Yes, the tiny model was working good.

**Comment by 459737087** [comment]
I also met this problem. I consider whether my dataset is wrong.
beacase there is many huge pixel, and they are larger than 1200.
